.github/workflows/codeql.yml

```diff
@@ -0,0 +1,25 @@
+name: CodeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: "18 6 * * 3"
+
+jobs:
+  analyze:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
```

Comment:
Had a look at this CodeQL thing since I didn't know it, and the documentation says these 2 are only required for private repositories.

Reply:
Good spot! Removed.

https://github.com/github/codeql-action#codeql-action